src/main/java/frc/team2412/robot/subsystems/IntakeSubsystem.java

```diff
@@ -81,18 +87,26 @@ public IntakeSubsystem() {
 		feederMotor = new CANSparkFlex(FEEDER_MOTOR, MotorType.kBrushless);
 
 		indexSensor = indexMotorUpper.getForwardLimitSwitch(SparkLimitSwitch.Type.kNormallyOpen);
-		feederSensor = new DigitalInput(FEEDER_SENSOR);
+		feederSensor = feederMotor.getForwardLimitSwitch(SparkLimitSwitch.Type.kNormallyOpen);
 
 		intakeFrontSensor = intakeMotorFront.getForwardLimitSwitch(SparkLimitSwitch.Type.kNormallyOpen);
 		// intakeBackSensor =
 		// intakeMotorBack.getForwardLimitSwitch(SparkLimitSwitch.Type.kNormallyOpen);
 		intakeLeftSensor = intakeMotorLeft.getForwardLimitSwitch(SparkLimitSwitch.Type.kNormallyOpen);
 		intakeRightSensor = intakeMotorRight.getForwardLimitSwitch(SparkLimitSwitch.Type.kNormallyOpen);
 
+		// sensor must be true for 0.1 seconds before being actually true
+		intakeFrontSensorDebouncer = new Debouncer(0.1);
+		intakeRightSensorDebouncer = new Debouncer(0.1);
+		intakeLeftSensorDebouncer = new Debouncer(0.1);
+
+		feederSensorDebouncer = new Debouncer(0.1);
```

Comment:
I thought one issue was the sensor briefly reading 'false' even when a note was held. If that's true, then per https://docs.wpilib.org/en/stable/docs/software/advanced-controls/filters/debouncer.html this should pass in a 2nd argument to specify that both the rising (note detected) & falling (note no longer detected) cases should be debounced.

Reply:
oops i missed that